File: docs/.validation/csharp/session-persistence_24.cs
// Source: guides/session-persistence.md:87
using GitHub.Copilot.SDK;

var client = new CopilotClient();

// Create a session with a meaningful ID
var session = await client.CreateSessionAsync(new SessionConfig
{
    SessionId = "user-123-task-456",
    Model = "gpt-5.2-codex",
});

// Do some work...
await session.SendAndWaitAsync(new MessageOptions { Prompt = "Analyze my codebase" });

// Session state is automatically persisted

File: docs/.validation/python/session-persistence_24.py
# Source: guides/session-persistence.md:48
import asyncio

async def main():
    from copilot import CopilotClient
    
    client = CopilotClient()
    await client.start()
    
    # Create a session with a meaningful ID
    session = await client.create_session({
        "session_id": "user-123-task-456",
        "model": "gpt-5.2-codex",
    })
    
    # Do some work...
    await session.send_and_wait({"prompt": "Analyze my codebase"})
    
    # Session state is automatically persisted

asyncio.run(main())